File: backend/src/main/java/dtos/ProductDTO.java
package dtos;

import java.io.Serializable;

import javax.xml.bind.annotation.XmlRootElement;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

/**
 * Product information type that the frontend consumes and produces.
 * 
 * @author Wanderley Drumond
 */
@XmlRootElement
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
public class ProductDTO implements Serializable {
	private Short id, totalLikes;
	private String name, image, section, price;
    private boolean hasLoggedUserLiked, hasLoggedUserFavorited;
	/**
	 * <p>The serial version identifier for this class.<p>
	 * <p>This identifier is used during deserialisation to verify that the sender and receiver of a serialised object have loaded classes for that object that are compatible with respect to serialisation.<p>
	 */
	private static final long serialVersionUID = 1L;
}

File: frontend/src/main/webapp/js/dtos/product_dto.js
export class ProductDTO {
    id;
    totalLikes;
    name;
    image;
    section;
    price;
    hasLoggedUserLiked;
    hasLoggedUserLikedFavorited;

    constructor() {};
    constructor(name, image, section, price) {
        this.name = name;
        this.image = image;
        this.section = section;
        this.price = price;
    };
}